src/gluonts/dataset/split.py

```diff
@@ -306,8 +306,12 @@ def test_pair(
         self, entry: DataEntry, prediction_length: int, offset: int = 0
     ) -> Tuple[DataEntry, DataEntry]:
         offset_ = self.offset + offset
-        if self.offset < 0 and offset_ >= 0:
-            offset_ += len(entry)
```

Comment:
This was clearly a bug, but I'm wondering whether `offset_ += entry[FieldName.TARGET].shape[-1]` is still needed here: if `self.offset == -offset` then `offset_ == 0` then the `input_slice` below may be incorrect? It would be `slice(None, 0)` which would be empty I think.

Reply:
I think it is not needed. If `self.offset == -offset` and `offset_ == 0` then the assertion fails since `-offset_ < prediction_length`. If we would add `offset_ += entry[FieldName.TARGET].shape[-1]` then the assertion would fail as well since `-offset_ < prediction_length` again. I can't think of any scenario where it would change the outcome.

Reply:
Right. I'm wondering though whether ensuring `offset_` is always positive would render the logic simpler to read.
So,

```python
if self.offset < 0:
    offset_ += entry[FieldName.TARGET].shape[-1]

assert offset_ + prediction_length <= entry[FieldName.TARGET].shape[-1], "Offset too short"
```

(didn't ☝️ check this thoroughly but makes sense to me)

Reply:
The two inequalities are equal then. I tested it, it works. I updated it accordingly.